CMakeLists.txt

```diff
@@ -108,6 +111,9 @@ set(LINK_LIBS
         zfp
         zstd
         tbb
+        stdc++fs
```

Comment:
For Mac, the `filesystem` is already available. This line will cause a building error. Maybe can move it to the lin 170 as 
```
if (${CMAKE_SYSTEM_NAME} MATCHES "Linux")
    target_link_libraries(carta_backend stdc++fs ${LINK_LIBS})
endif (${CMAKE_SYSTEM_NAME} MATCHES "Linux")
```

Reply:
> For Mac, the `filesystem` is already available. This line will cause a building error. Maybe can move it to the lin 170 as 
> ```
> if (${CMAKE_SYSTEM_NAME} MATCHES "Linux")
>     target_link_libraries(carta_backend stdc++fs ${LINK_LIBS})
> endif (${CMAKE_SYSTEM_NAME} MATCHES "Linux")
> ```

Good point. I'll push a fix along those lines. It might be more related to clang vs GCC, rather than linux versus Darwin

Reply:
I've added a `if (CMAKE_CXX_COMPILER_ID MATCHES "GNU")` switch to `CMakeLists.txt` to only include `stdc++fs` when using `GCC`